### PR TITLE
Add a method to ModelTestCase for use without config files

### PR DESCRIPTION
**Pull request by matt-gardner**
This was necessary to have tests for the no-config-file template repo.  I also noticed a few other small typos in the documentation, which are fixed in here.

### CODE REVIEW

File: allennlp/common/testing/model_test_case.py [hunk 1]
```diff
@@ -177,6 +178,63 @@ def ensure_model_can_train_save_and_load(
...
+        A simple test for model training behavior when you are not using configuration files. In
+        this case, we don't have a story around saving and loading models (you need to handle that
+        yourself), so we don't have tests for that.  We just test that the model can train, and that
+        it computes gradients for all parameters.
+
+        Because the `Trainer` already has a reference to a model and to a data loader, we just take
+        the `Trainer` object itself, and grab the `Model` and other necessary objects from there.
+
+        # Parameters
+
```
Comment (epwalsh) [inline allennlp/common/testing/model_test_case.py:200]: Missing `trainer` param here

Comment (matt-gardner) [reply allennlp/common/testing/model_test_case.py:200]: Thanks, good catch.

**Comment by matt-gardner** [comment]
@epwalsh, this is now failing docs builds, and I am having a really hard time figuring out what the offending line is.  Any help?  Is there a way to have the scripts actually output the line that's the problem?

**Comment by epwalsh** [comment]
That’s weird, I’ve never seen that before. I’ll try to debug that on Monday

**Pull request merged**